docusaurus/docs/dev-docs/migration/v4-to-v5/guides/plugins-migration.md

```diff
@@ -0,0 +1,51 @@
+---
+title: Plugins migration reference
+displayed_sidebar: devDocsMigrationV5Sidebar
+---
+
+# Plugins migration summary
+
+:::callout 🚧  Work in progress
+This page is a work-in-progress and some links are currently missing.
+:::
+
+The present page is intended to be used as a short summary of everything to consider if you are a plugin developer intending to migrate your plugin from Strapi v4 to Strapi 5. The page quickly describes the changes affecting plugins and link to additional resources where necessary.
+
+## Back-end changes
+
+- The Entity Service API from Strapi v4 is deprecated and Strapi 5 uses the [Document Service API](/dev-docs/api/document-service) instead. A [migration guide](/dev-docs/migration/v4-to-v5/guides/from-entity-service-to-document-service) is available to help you transition to the Document Service API.
+- General Strapi v4 to Strapi 5 [breaking changes](/dev-docs/migration/v4-to-v5/breaking-changes) might apply.
+- You can use the new [Plugin CLI](/dev-docs/plugins/guides/use-the-plugin-cli) to generate plugins and publish them on NPM and/or submit them to the Marketplace.
+
+## Front-end changes
+
+- The Design System is upgraded to v2 in Strapi 5:
+  <!-- TODO: add link to icons in Design System v2 -->
+  - There are no big visual changes, except for [icons](#).<br/>(_the link will be added in the upcoming weeks_)
+  <!-- TODO: add link to migration guide -->
+  - A migration guide will be [available](#).<br/>(_the link will be added in the upcoming weeks_)
+  <!-- TODO: add link to breaking changes -->
+  - A list of breaking changes specific to the Design System will be available in the [Design System documentation](#).<br/>(_the link will be added in the upcoming weeks_)
+  - General Strapi v4 to Strapi 5 [breaking changes](/dev-docs/migration/v4-to-v5/breaking-changes) might apply.
+- The `helper-plugin` is deprecated. A [migration reference](/dev-docs/migration/v4-to-v5/guides/helper-plugin) is available to help you transition away from the `helper-plugin`.
+  <!-- TODO: clarify the next line 👇 -->
+  <!-- - Aliasing dependencies → if not declared it will use ours (probably) -->
+
+<!-- TODO: clarify these 👇-->
+<!-- ## General changes
+
+- Building and packaging?
+  - They don’t have to do it (not tested)
+  - Recommended as a best practices (from npm)
+- Pack up?
+  - Not required
+  - Is pack up specific to our packages or is it universal
+      - More for libraries
+  - Does the plugin need to be a TS one to use pack up (honestly no idea what pack up does)
+- peerDepend requirement?
+  - Yes probably (ask emilie)
+  - As a peerDepend -->
```

Comment:
This is up to you if you want to write about these bits – the building/packaging is best practices for all libraries on NPM not isolated to Strapi views, using pack-up again, if you write about the former you can suggest using the plugin-cli to build your plugins, i'm not sure where the assumption came in that we might force you to use `pack-up`, but we really don't care at the end of the day.

The peerDep rule for `@strapi/strapi` it's highly recommended and most likely needed by the library because they'll access imports from that package... I'm not sure there's a difference from V4...

Reply:
Thanks, Josh! I will strip up this content from the comments, then. If something is unclear or missing we'll add it later.